src/state/slices/portfolioSlice/portfolioSlice.ts

```diff
@@ -311,37 +354,53 @@ export const selectPortfolioTotalFiatBalance = createSelector(
       .toFixed(2)
 )
 
-const selectAssetIdParam = (_state: ReduxState, id: CAIP19) => id
-const selectCAIP10Param = (_state: ReduxState, id: CAIP10) => id
-
-export const selectPortfolioFiatBalanceById = createSelector(
+export const selectPortfolioFiatBalanceByAssetId = createSelector(
   selectPortfolioFiatBalances,
   selectAssetIdParam,
   (portfolioFiatBalances, assetId) => portfolioFiatBalances[assetId]
 )
 
-export const selectPortfolioCryptoBalanceById = createSelector(
-  selectPortfolioBalances,
+export const selectPortfolioFiatBalancesByFilter = createSelector(
+  selectPortfolioFiatBalances,
+  selectPortfolioFiatAccountBalances,
+  selectAssetIdParamFromFilter,
+  selectAccountIdParamFromFilter,
+  (portfolioAssetFiatBalances, portfolioAccountFiatbalances, assetId, accountId) => {
+    if (assetId && !accountId) return portfolioAssetFiatBalances[assetId]
+    if (assetId && accountId) return portfolioAccountFiatbalances[accountId][assetId]
+    if (!assetId && accountId) {
+      const accountBalances = portfolioAccountFiatbalances[accountId]
+      const totalAccountBalances = Object.values(accountBalances).reduce(
+        (totalBalance: string, fiatBalance: string) => {
+          return bnOrZero(totalBalance).plus(fiatBalance).toFixed(2)
+        },
+        '0'
+      )
+      return totalAccountBalances
+    }
+    return ''
```

Comment:
should this return `'0'` rather than an empty string, so it's kind of consistent? `'0'` is also falsy

```
Welcome to Node.js v14.17.3.
Type ".help" for more information.
> '0' == true
false
```

Reply:
disregard this i have brain damage